MWT/architecture.rst

```diff
@@ -0,0 +1,99 @@
+Architecture
+============
+
+Structure
+---------
+
+.. figure:: images/hierarchy.png
+   :alt: MWT Hierarchy
+   :align: center
+
+Widget
+~~~~~~
+
+A widget is an object that is intended to be displayed on a screen and that can interact with the user. A Widget occupies a specific region of the display and holds a state.
+
+Widgets are arranged on a desktop. A widget can be part of only one desktop hierarchy, and can appear only once on that desktop.
+
+Container
+~~~~~~~~~
+
+A container follows the composite pattern: it is a widget composed of other widgets. It also defines the layout policy of its children (defining their bounds). The children's locations are relative to the location of their parent.
+Containers can be nested to design elaborate user interfaces.
+
+Widgets added to a container are stored in a list. The order of the list defines the widgets' front-to-back stacking order within the container. By default, a Widget is added to the end of the list (so to the bottom of the stacking order).
```

Comment:
this is relevant only for overlapping containers right?

Reply:
Yes. But it is still interesting to know when writing a container.

Reply:
i think this is not very intuitive because there is no back or front in most containers. since they lay out their children in a way that two children can not overlap.

i would say instead that the children are rendered in the order in which they have been added to the children list. and thus if the container allows overlapping, the widgets added last will be on top of the widgets added first.

i think it is easier to understand by talking about the most common case first, and then the special case (overlap container)

Reply:
Fixed.

Reply:
thanks :P